src/widget/findonwebmenufactory.h

```diff
@@ -0,0 +1,11 @@
+#pragma once
+
+#include <QMenu>
+
+class Track;
+
+class FindOnWebMenuFactory {
+  public:
+    static void createFindOnWebSubmenus(QMenu* pFindOnWebMenu,
+            const Track& track);
+};
```

Comment:
Thanks for reviewing and giving feedback :+1: 

I have a little question. Should the class and file be removed or moved to WFindOnMenuWeb menu or WTrackMenu then? I think it can be extended with the preferences option in this class and file. Users can choose which services will be populated later on if this option is implemented.

Reply:
IMO just keep it where it is. The point was just that we don't need unnecessary classes.

```suggestion
namespace mixxx {
namespace library {
void createFindOnWebSubmenus(QMenu* pFindOnWebMenu, const Track& track);
}
}
```

The problem with that is that it changes some visibility rules in regards to linkage which I don't fully understand. @daschuer what do you think about this proposal?

Reply:
I don't really mind. The current implementations visualizes the factory pattern better. Static classes is the traditional solution for grouping free functions. Since this is what a namespace does, using it for this purpose is more "correct".
So I suggest to follow @Swiftb0y here.

Reply:
Done :+1:, I hope It's correct.

Reply:
Should be correct as long as it compiles.

Reply:
Yeap, It compiles. I compile few times (just making sure hehe) before every commit :sweat_smile: